jdaviz/core/region_translators.py

```diff
@@ -353,6 +356,214 @@ def aperture2regions(aperture):
     return region_shape
 
 
+def _create_polygon_skyregion_from_coords(coords, frame='icrs', unit=u.deg):
+    """Create a `regions.PolygonSkyRegion` from given coordinates.
+
+    Parameters
+    ----------
+    coords : list
+        List of coordinates in the form of [ra1, dec1, ra2, dec2, ...].
+
+    frame : str, optional
+        Coordinate frame. Default is 'icrs'.
+
+    unit : `~astropy.units.Unit`, optional
+        Unit of the coordinates. Default is `~astropy.units.deg`.
+
+    Returns
+    -------
+    sky_region : `regions.PolygonSkyRegion`
+        A polygon sky region.
+
+    Examples
+    --------
+    Create a `regions.PolygonSkyRegion` from given coordinates:
+
+    >>> from jdaviz.core.region_translators import create_polygon_skyregion_from_coords
+    >>> coords = [10.0, 20.0, 30.0, 40.0, 50.0, 60.0]
+    >>> _create_polygon_skyregion_from_coords(coords)
+    <PolygonSkyRegion(vertices=[<SkyCoord (ICRS): (ra, dec) in deg
+        (10., 20.), (30., 40.), (50., 60.)>])>
+
+    """
+    ra = np.array(coords[::2], dtype=float)
+    dec = np.array(coords[1::2], dtype=float)
+    sky_coordinates = SkyCoord(ra, dec, frame=frame, unit=unit)
+    sky_region = PolygonSkyRegion(sky_coordinates)
+
+    # Need these for zooming
+    length = sky_coordinates[0].separation(sky_coordinates[1])
+    width = sky_coordinates[1].separation(sky_coordinates[2])
+    sky_region.height = Angle(max(length, width), unit)
+    sky_region.center = SkyCoord(ra.mean(), dec.mean(), unit=unit)

```

Comment:
Let's keep these specific lines in `jwst_header_to_skyregion` in `slit_overlay.py`, since they're not always relevant for general polygons.